### PR TITLE
monitor: support using idf_monitor from a virtualenv on Windows

### PR DESCRIPTION
**Pull request by ajf58**
Ensure that the venv's Python executable is used, and convert the WindowsPath to a string.

### CODE REVIEW

**Review by sylvioalves — APPROVED** [review]

File: west/tools.py [hunk 1]
```diff
@@ -181,8 +181,8 @@ def monitor(self, module_path, args):
         monitor_path = Path(module_path, "tools/idf_monitor/idf_monitor.py")
         cmd_path = Path(os.getcwd())
         if platform.system() == 'Windows':
-            cmd_exec(("python.exe", monitor_path, "-p", esp_port,
-                     "-b", args.baud, elf_path, "--eol", args.eol), cwd=cmd_path)
+            cmd_exec((sys.executable, str(monitor_path), "-p", esp_port,
```
Comment (marekmatej) [inline west/tools.py:184]: It almost looks like the `platform.system()` condition could be removed, right? Can you make the `cmd_exec` call universal for all platforms?

**Review by marekmatej — CHANGES_REQUESTED** [review]
I'm wondering if we can unify both cases.

File: west/tools.py (hunk 1 above)
Comment (ajf58) [reply west/tools.py:184]: It might work, but I don't have a non-Windows host to try that change out with.

Comment (marekmatej) [reply west/tools.py:184]: @ajf58 from what I see here, the only differences between Windows and non-Windows cases are the stringification of 2 arguments. Therefore I asked if you remove the condition. I'll test it here and let you know.

Comment (ajf58) [reply west/tools.py]: OK, that's done. Please let me know how your testing gets on.

**Review by LucasTambor — APPROVED** [review]

**Pull request merged**